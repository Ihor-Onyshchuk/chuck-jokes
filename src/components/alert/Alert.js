import React from 'react';
import T from 'prop-types';

const Alert = ({type = 'info', text}) => (
  <div className={`text-center alert alert-${type}`} role="alert">
    {text}
  </div>
);

Alert.propsTypes = {
  type: T.string,
  text: T.string.isRequired,
};

export default Alert;
