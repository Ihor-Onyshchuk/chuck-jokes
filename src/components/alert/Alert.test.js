import React from 'react';

import {shallow} from 'enzyme';
import Alert from './Alert';

const text = 'Test text';
let wrapped = shallow(<Alert>{text}</Alert>);

describe('Alert', () => {
  it('should render the Alert Component correctly', () => {
    expect(wrapped).toMatchSnapshot();
  });
  it('renders the text', () => {
    expect(wrapped.find('div').children.length).toEqual(1);
  });
});
