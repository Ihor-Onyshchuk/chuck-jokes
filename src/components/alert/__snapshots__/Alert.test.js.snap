// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render the Alert Component correctly 1`] = `
<div
  className="text-center alert alert-info"
  role="alert"
/>
`;
